modules/tasks/tasks/src/main/java/org/motechproject/tasks/service/impl/TaskServiceImpl.java

```diff
@@ -525,7 +527,7 @@ private TaskDataProvider findProviderByName(DataSource ds) {
             errors.addAll(taskValidator.validateAction(action, channel));
             TriggerEvent trigger = triggerEventService.getTrigger(task.getTrigger());
             Map<Long, TaskDataProvider> providers = getProviders(task);
-            ActionEvent actionEvent = channel.getAction(action);
+            ActionEvent actionEvent = ChannelUtils.getAction(channel, action);
```

Comment:
This was done to solve circular dependencies. I couldn't think of anything more appropriate at that time.